public/js/all.js

```diff
@@ -63,136 +43,31 @@ function strPad(input, padLength, padString) {
-    return;
+function onSubmitComment(event) {
+  event.preventDefault();
+  var $form = $(event.currentTarget);
+  var $submit = $form.find('.comment-submit-button');
+  var $error = $form.find('.form-error');
+
+  var input = $form.serializeArray().reduce(function (obj, item) {
+    obj[item.name] = item.value;
+    return obj;
+  }, {});
+
+  // validate
+  $error.hide();
+  if (input.c.length === 0) {
+    $error.show();
+    $error.text('Comment is empty');
+    return false;
   }
 
-  var sPath = window.location.pathname;
-  if (sPath.substr(0, 5).toLowerCase() === '/game') {
-    window.location.reload();
-    return;
-  }
-  if (sPath.substr(0, 12).toLowerCase() === '/achievement') {
-    window.location.reload();
-    return;
-  }
-  if (sPath.substr(0, 5).toLowerCase() === '/user') {
+  $submit.find('img').attr('src', '/Images/loading.gif');
+  $.post('/request/comment/create.php', input).done(function () {
     window.location.reload();
-    return;
-  }
-
-  var commentRow = $('#comment_art_' + articleType + '_' + data);
-  if (commentRow.length) {
-    // Embed as proper comment instead!
-    commentRow.addClass('feed_comment');
-    commentRow.removeAttr('id');
-    var textBox = commentRow.find('#commentTextarea');
-    if (textBox.length) {
-      var comment = textBox.val();
-      // var safeComment = comment.replace( /<|>/g, '_' );
-      var safeComment = stripTags(comment);
-      // var safeComment = comment; // Removed!
-
-      textBox.after(safeComment);
-      // Set its container to commenttext
-      textBox.parent().addClass('commenttext');
-      textBox.remove();
-    }
-
-    var submitButton = commentRow.find('#submitButton');
-    if (submitButton.length) {
-      submitButton.remove();
-    }
-  }
-}
+  });
 
-function processComment(activityVar, articleType) {
-  var user = readCookie('RA_User');
-  if (user !== null) {
-    var rowID = 'comment_' + activityVar;
-    var commentRow = $('#' + rowID);
-    if (commentRow.length) {
-      var textBox = commentRow.find('#commentTextarea');
-      if (textBox.length) {
-        var comment = textBox.val();
-        if (comment.length > 0) {
-          var safeComment = stripTags(comment);
-          // var safeComment = comment.replace( /<|>/g, '_' );
-          // Note: using substr on activityVar, because it will be in the format art_3_213 etc.
-          var activityId = activityVar.substr(activityVar.lastIndexOf('_') + 1);
-          var posting = $.post('/request/comment/create.php', {
-            a: activityId,
-            c: safeComment,
-            t: articleType,
-          });
-          posting.done(function (data) { onCommentSuccess(data, articleType); });
-          var submitButton = commentRow.find('#submitButton');
-          if (submitButton.length) {
-            submitButton.attr('src', '/Images/loading.gif'); // Change to 'loading' gif
-            submitButton.attr('onclick', ''); // stop being able to click this
```

Comment:
I was able to submit the same comment multiple times by accidentally clicking on the spinner

Reply:
Oh my, good call. The submit button is now removed and fully replaced with the loading spinner.
This has the advantage that the loading image is properly prefixed with the `ASSET_URL`.